### PR TITLE
Fix hide sell at market button when durationunit is tick

### PR DESCRIPTION
**Pull request by DanSallau**
Hi,

Attached contained few changes that should fix our or hide sell at market button when the duration unit is ticks. Should there be any issue with regard to this, i would be glad to address. Trello https://trello.com/c/0aPKfajT/31-sell-at-market-shouldn-t-be-shown-for-trades-with-duration-as-ticks

### CODE REVIEW

File: src/barrier-picker/SpreadBarrierCard.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default class SpreadBarrierCard extends Component {
                 <InputGroup
                     type="number"
                     label={`Amount per point (${currency})`}
-                    value={spreadInfo.amountPerPoint}
+                    defaultValue={spreadInfo.amountPerPoint}
```
Comment (qingweibinary) [inline src/barrier-picker/SpreadBarrierCard.js]: It uses value for a reason, to preserve state when change mode, we need to use value

File: src/contract-details/ContractDetailsCard.js [hunk 2]
```diff
@@ -48,24 +49,25 @@ export default class ContractDetailsCard extends Component {
...
-						className="sell-at-market-btn"
-						onClick={() => actions.sellContract(contract.contract_id, 0)}
-					>
-						Sell at Market (<NumberPlain
-							value={contract.bid_price}
-							currency={contract.currency}
-							isProfit={v => v - contract.buy_price}
-						/>)
-					</button> :
+				{(durationUnit !== 't') && (
```
Comment (qingweibinary) [inline src/contract-details/ContractDetailsCard.js]: Have you check that this actually happens?

By right, if it's not allowed to be sold, is_valid_to_sell would be 0, thus validToSell would be false, 

It should not required an additional field check for this.

**Comment by qingweibinary** [comment]
I've checked, it's a backend issue, our current code is doing fine as long as we check on `is_valid_to_sell`
IMO, we should not decide the business rule in frontend unless necessary

Trello card of issue
https://trello.com/c/RZyeHDWh

File: src/barrier-picker/SpreadBarrierCard.js (hunk 1 above)
Comment (nuruddeensalihu) [inline src/barrier-picker/SpreadBarrierCard.js]: Using value would fix the value though and therefore close it for modification or doesn't it ? It it need to use value why not use label ? Or am i confusing something ?

Comment (qingweibinary) [inline src/barrier-picker/SpreadBarrierCard.js]: Actions are used to change the value ~
We can bind actions on the input

File: src/contract-details/ContractDetailsCard.js (hunk 2 above)
Comment (nuruddeensalihu) [inline src/contract-details/ContractDetailsCard.js]: So if durationUnit is of tick value then , is_valid_to_sell would be 0 , is that what you are insinuating ? If that's the case i think its from the backend perhaps. The (durationUnit !== 't') is to check explicitly, with that i don't think sell on market would appear not even for a second .

File: src/barrier-picker/SpreadBarrierCard.js (hunk 1 above)
Comment (nuruddeensalihu) [inline src/barrier-picker/SpreadBarrierCard.js]: Actions are used to change the value and from the same input group whose value has been fixed? Or do we have a different UI for that  ?

File: src/contract-details/ContractDetailsCard.js (hunk 2 above)
Comment (qingweibinary) [inline src/contract-details/ContractDetailsCard.js]: Yes ~
it's from backend and should be from backend, the durationUnit check would be some temporary workaround, as ContractDetail should not be deciding what to render by itself

File: src/barrier-picker/SpreadBarrierCard.js (hunk 1 above)
Comment (qingweibinary) [inline src/barrier-picker/SpreadBarrierCard.js]: Have a look on other barrier?

Specifically the onStopLossChange and onStopProfitChange are actions that suppose to change the value of Input

Comment (nuruddeensalihu) [inline src/barrier-picker/SpreadBarrierCard.js]: I have seen all those. The value have to changed before they are fired right ? How can the value change when you fix the value by setting the value ?  Check again .

File: src/contract-details/ContractDetailsCard.js (hunk 2 above)
Comment (borisyankov) [inline src/contract-details/ContractDetailsCard.js]: After we start having these conditionals, even if they are correct, we need to extract to a 'biz logic' function/lib somewhere + unit test them

**Pull request closed without merging**